js/modules/modules.go

```diff
@@ -108,8 +107,9 @@ func getInterfaceMethods() []string {
 	return result
 }
 
-// InstanceCore is something that will be provided to modules and they need to embed it in ModuleInstance
-type InstanceCore interface {
+// InstanceParams is something that will be provided to module instance on initialization so that it can get access to
+// everything it needs
+type InstanceParams interface {
```

Comment:
Looking for a better name, after which I will rebase it to get rid fixup commits

Reply:
Following the Go naming convention for interfaces, you would end up with something like `ContextRuntimeGetter`. Idiom aside, I find `InstanceParams` explicit enough, although I didn't expect/realize it was an interface until I checked how it was used: when seeing the `InstanceParams` argument in the `NewModuleInstance` it wasn't immediately clear it was an interface.

Hope that's helpful 👀

Reply:
The name seems fine, and is more in line with output extensions. :+1:

I'd just make the comment more descriptive. Maybe:
```
InstanceParams will be provided to module instances on initialization so that they can access internal k6 objects.
```

Reply:
`ModuleParams` ?

Reply:
@oleiade the problem is that it will not only be about Getting those 2 or even the actual 4 it''s giving now. 

In #2228 this is also the way to add something to the event loop. 

In practice it's interacting with k6, predominantly in the context of 1 VU and the current iteration. So something like `State` would've made sense except for the part that `modules.State.AddToEventLoop` sounds bad.

_tangent: `AddToEventLoop` might not be the end method that will be used btw, it might be what currently is named `Reserve` on the eventloop, which in practice returns a function taking a function (`func(func())` where the idea is that it needs to be called to add something to the event loop but at the same time it will block the event loop from ending even if the loop is empty. (It doesn't actually reserve a position in the queue or something like that). Given that the name for that might be `modules.State.blockEventLoopUntil` or ... something actually better :rofl: 

@imiric 
How about?
> InstanceParams will be provided to module instances on initialization so that they can access interact with k6.

Again the eventLoop and the likely additional stuff for an module/VU lifecycle integration will be at least in part, part of this

Reply:
@yorugac the full name currently is `modules.InstanceParams` and these are the params for when an *instance* of the module on each `import "nameofmodule"` is being evaluated. There currently are *no* params provided just when we load "nameofmodule" because it's part of the k6 build but isn't actually instanciated.

Reply:
`InstanceParams` is suitable for the current design. But the thing is, this is not actually an instance's parameters. It's for providing a way for a module to access a VU's whereabouts.

I won't say we should call it `VUWhereabouts` or `VUUniverse` (_you can type it with more Us_) 😄  Maybe, it can be something like `VUInternals`.

Reply:
`VUInternals` seems like the best option so far

Reply:
Yes, @MStoykov I meant this `params` weren't about `instance` per se... So I'd say :+1: for `VUInternals` too but: event loop is going to add methods for promises there -- so it won't be `internals` either
How large would that interface grow?

Reply:
>  How large would that interface grow?

Given that it provides a way for the modules to interact with k6, as much as we need it to interact :). The alternative to this growing is to keep adding interfaces that either it implements or the `modules.Instance` implements so that we can call something on it :shrug: